packages/compass-connections/src/hooks/use-connection-repository.spec.ts

```diff
@@ -0,0 +1,210 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import { useConnectionRepository } from './use-connection-repository';
+import chai, { expect } from 'chai';
+import { stub } from 'sinon';
+import {
+  type ConnectionStorage,
+  ConnectionStorageBus,
+} from '@mongodb-js/connection-storage/renderer';
+import { ConnectionStorageContext } from '@mongodb-js/connection-storage/provider';
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor } from '@testing-library/react';
+import { createElement } from 'react';
+import chaiAsPromised from 'chai-as-promised';
+
+chai.use(chaiAsPromised);
```

Comment:
I don't see any special promise expectations from this packagebeing used here, why are we adding this?

Reply:
It is used in `should not save a new connection if it has an invalid connection string`:

```js
await expect(
        result.current.saveConnection({
          id: '1',
          connectionOptions: { connectionString: 'mongo://table.row:8080' },
        })
      ).to.be.rejected;
```

Reply:
Ah, missed that, got it! Feel free to keep it, FYI this package was causing tests hanging issues before (due to kept promise references that would now allow the mocha process to end), so most of the tests are testing error rejection with normal expects now, something like:

```js
try {
  await fnThatShouldThrow();
  expect.fail("Expected fnThatShouldThrow to throw")
} catch (err) {
  expect(err).to.have.property('message', 'Expected error message')
}

That way you are actually testing that the error you expect was thrown, and not a random one, and avoiding an extra dep
```

Reply:
Ah I can drop it then and change it to a catch, tests are already passing so I can just do the change, verify this test pass and merge.

Reply:
Only if you want to! I legit just didn't notice the usage and remember why we kinda stopped using it (but there are still a few places left)

Reply:
I think if we had problems with the library I prefer to remove it just in case it doesn't happen again :D. It's a pretty straightforward change so no worries.

Thanks for pointing it out!